### PR TITLE
Containerless support for Java apps

### PR DESCRIPTION
**Pull request by eemcmullan**
See docs/containerless.md for testing instructions

### CODE REVIEW

File: cmd/analyze.go [hunk 1]
```diff
@@ -70,6 +68,17 @@ var (
 	}
 )
 
+// analyzer container paths
+const (
+	RulesetPath            = "/opt/rulesets"
```
Comment (shawn-hurley) [inline cmd/analyze.go:73]: What happens with these paths on windows?

**Review by shawn-hurley — CHANGES_REQUESTED** [review]
Can we just run the static report from the main process rather than call the binary via script? IIUC what is happening

File: cmd/analyze.go (hunk 1 above)
Comment (eemcmullan) [reply cmd/analyze.go:73]: These paths are only used by the analyzer container, which are linux containers anyways. They aren't used in the containerless work

File: cmd/analyze-bin.go [hunk 2]
```diff
@@ -0,0 +1,842 @@
...
+				os.Exit(1)
+			}
+			err = analyzeBinCmd.Validate(cmd.Context())
+			if err != nil {
+				log.Error(err, "failed to validate flags")
+				return err
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
```
Comment (jmle) [inline cmd/analyze-bin.go:103]: Is there any possibility of breaking this method down?

File: cmd/analyze-bin.go [hunk 3]
```diff
@@ -0,0 +1,842 @@
...
+					return fmt.Errorf("must specify only one input source")
+				}
+			}
+		}
+		stat, err := os.Stat(b.input)
+		if err != nil {
+			return fmt.Errorf("%w failed to stat input path %s", err, b.input)
+		}
+		// when input isn't a dir, it's pointing to a binary
+		// we need abs path to mount the file correctly
```
Comment (jmle) [inline cmd/analyze-bin.go:432]: Is this comment still relevant here?

**Comment by eemcmullan** [comment]
> Can we just run the static report from the main process rather than call the binary via script? IIUC what is happening

@shawn-hurley The first step of static report does run from main, which creates the output.js file. The script is there just because the second step required in generating the static report, building the dir and moving over the output.js file and then moving that dir into user output, requires a decent amount of chaining commands. Just thought the script made things more straight-forward, but could be wrong.

File: cmd/analyze-bin.go (hunk 3 above)
Comment (eemcmullan) [reply cmd/analyze-bin.go:432]: We're not. Good catch

**Review by aufi — APPROVED** [review]
Overall looks good and mostly fits to the https://github.com/konveyor/enhancements/pull/196, LGTM.

File: static-report.sh [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+if [ "$(uname)" == "Linux" ] || [ "$(uname)" == "Darwin" ]; then
+  HOMEDIR="$HOME"
+else
+  # windows
+  HOMEDIR="%USERPROFILE%"
+fi
+
+# TODO test this on windows
+(cd ${HOMEDIR}/.kantra/static-report && 
+npm clean-install &&
```
Comment (pranavgaikwad) [inline static-report.sh:15]: if I am not wrong, this is happening at runtime. I don't think we can expect the users to have npm installed. This should be a built artifact that's already present in `.kantra/static-report`. we just have to copy it into the output dir and put output.js under the copied dir.

**Review by pranavgaikwad — CHANGES_REQUESTED** [review]
Overall, code makes sense. Concern about static-report..

File: static-report.sh (hunk 4 above)
Comment (eemcmullan) [reply static-report.sh:15]: @pranavgaikwad the `build` dir doesn't have to be created for each new analysis output? Sorry if that's not what you're talking about. I wasn't sure

Comment (pranavgaikwad) [reply static-report.sh:15]: yeah build dir is a one time process when we create a static-report release. The contents of the build directory remain constant for different analyses, the only variable is output.js file.

Comment (eemcmullan) [reply static-report.sh:15]: Gotcha. Apologies - misunderstood that step. Will update

**Review by pranavgaikwad — APPROVED** [review]

**Review by shawn-hurley — APPROVED** [review]

**Pull request merged**